putiopy.py

```diff
@@ -383,9 +387,13 @@ def _verify_file(self, filepath):
         logger.info('crc OK')
         return True
 
-    def _download_file(self, dest, delete_after_download, chunk_size):
+    def _download_file(self, dest, delete_after_download, max_kbps, chunk_size):
         name = _str(self.name)
 
+        if max_kbps:
+            token_bucket = TokenBucket()
+            token_bucket.set_rate(max_kbps)
+
```

Comment:
You are creating a new bucket for every file download. If there are many small files this will not be effective. Can you make the bucket a property of client?

Reply:
Yeah. Give me a day or two.